Logger/Record.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using static System.Runtime.InteropServices.OptionalAttribute;
+namespace Logger;
+
+
+/*
+ The reason we implemented our FullName record in a class
+called records is so we could keep all of our records together.
+It makes it easier to navigate and understand our program
+from an outside perspective.  
+*/
+public class Record
+{
+    /*
+    We decided to do a reference type because it is better for storing more
+    values. Our record for FullName is mutable because names can be
+    occasionally changed. Exp. Changing last name after marriage.  
+    */
+    public record FullName(string? firstName, string? lastName, string? middleName)
+    {
+        public string firstName { get; } = firstName ?? throw new ArgumentNullException(nameof(firstName));
+        public string lastName { get; } = lastName ?? throw new ArgumentNullException(nameof(lastName));
+        public string middleName { get; } = middleName ?? throw new ArgumentNullException(nameof(middleName));
+
+    }
+    //implicitly overrding Name in Entity since we are only using Name.
+    public record Person(FullName FullName) : Entity
+    {
+         public override string Name { get => FullName.ToString(); }
```

Comment:
```suggestion
         public override string Name { get => FullName.ToString(); }
```
What would you expect the `ToString` method to return here? You should consider adding a test for this.